runtime/str.go

```diff
@@ -205,6 +205,45 @@ func strEq(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return strCompare(v, w, False, True, False), nil
 }
 
+func strFind(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	// TODO: Support finding index of unicode substring.
+	expectedTypes := []*Type{StrType, StrType, ObjectType, ObjectType}
+	argc := len(args)
+	if argc == 2 || argc == 3 {
+		expectedTypes = expectedTypes[:argc]
+	}
+	if raised := checkMethodArgs(f, "find/index", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	s := toStrUnsafe(args[0]).Value()
+	l := len(s)
+	start, end := 0, l
+	if argc >= 3 {
+		startParsed, raised := adjustBounds(f, args[2], l, false)
```

Comment:
Maybe declare raised above and then just assign directly to start to avoid the extra startParsed temporary:

```
var raised *BaseException
if argc >= 3 {
  start, raised = ...
}
```